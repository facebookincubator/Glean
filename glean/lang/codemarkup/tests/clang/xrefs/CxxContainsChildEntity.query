query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
