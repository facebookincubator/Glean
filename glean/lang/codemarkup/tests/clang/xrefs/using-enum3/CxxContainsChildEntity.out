[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N1" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 9,
                "columnBegin": 1,
                "lineEnd": 13,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "enum_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "E" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N1" } } }
                  }
                }
              },
              "isScoped": false,
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 11,
                "columnBegin": 1,
                "lineEnd": 11,
                "columnEnd": 18
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N21" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 15,
                "columnBegin": 1,
                "lineEnd": 24,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N21" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 20,
                "columnBegin": 1,
                "lineEnd": 22,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N22" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 26,
                "columnBegin": 1,
                "lineEnd": 35,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N22" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 31,
                "columnBegin": 1,
                "lineEnd": 33,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N31" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 37,
                "columnBegin": 1,
                "lineEnd": 46,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N31" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 42,
                "columnBegin": 1,
                "lineEnd": 44,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N32" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 48,
                "columnBegin": 1,
                "lineEnd": 57,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N32" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 53,
                "columnBegin": 1,
                "lineEnd": 55,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N41" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 59,
                "columnBegin": 1,
                "lineEnd": 68,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N41" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 64,
                "columnBegin": 1,
                "lineEnd": 66,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N42" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 70,
                "columnBegin": 1,
                "lineEnd": 79,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N42" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 75,
                "columnBegin": 1,
                "lineEnd": 77,
                "columnEnd": 1
              }
            }
          }
        }
      }
    }
  }
]